constraints/include/constraint_data.h

```diff
@@ -0,0 +1,37 @@
+//
+// Created by gavin on 6/27/24.
+//
+
+#ifndef CONSTRAINT_DATA_H
+#define CONSTRAINT_DATA_H
+
+#include <eigen3/Eigen/Dense>
+#include <eigen3/Eigen/Sparse>
+
+namespace torc::constraint {
+    enum CONSTRAINT_T {
+        Equals,         // equality constraint
+        LessThan,       // less than or equal to constraint
+        GreaterThan     // greater than or equal to constraint
+    };
+
+    /**
+     * Holds the matrices that the Constraint class will use to return values.
+     * @tparam scalar_t type of scalar to use in the constraint
+     */
+    template <class scalar_t>
+    struct ConstraintData {
```

Comment:
Oh sorry what I meant was:
A single struct for each QP type. i.e. one struct for lb <= Ax <= ub (so that struct holds lb, A, ub), another struct for Ax = b, Gx <=h (that struct holds A, b, G, h), etc...

Then each function can take in a reference to their associated struct.

Reply:
I see! Will change that right now